cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -231,7 +231,7 @@ func GetKubeletDir(dryRun bool) (string, error) {
 // backupAPIServerCertAndKey backups the old cert and key of kube-apiserver to a specified directory.
 func backupAPIServerCertAndKey(certAndKeyDir string) error {
 	subDir := filepath.Join(certAndKeyDir, "expired")
-	if err := os.Mkdir(subDir, 0766); err != nil {
+	if err := os.Mkdir(subDir, 0750); err != nil {
```

Comment:
given `subDir` is under `DefaultCertificatesDir = "/etc/kubernetes/pki"` which requires root and that this is not an executable, the permissions should be `0660`.

Reply:
@neolit123 I don't think, that we should remove the `x` bit for the owner (even though this does nothing for root). Also, why are you suggesting to add a `g+w` bit?

Reply:
@rosti 
group write permission for GID=0 seems reasonable, to me.

@ping035627 

> @neolit123 After I changed to 0660, there was no problem testing under root, but other users had problems. It reports "permission denied" when os.Rename. I think we should also use 0750.

which tests are failing?

Reply:
--- FAIL: TestBackupAPIServerCertAndKey (0.00s)
    postupgrade_test.go:54: Failed to backup cert and key in dir /tmp/978260827: couldn't move these files: map[]. Got errors: rename /tmp/978260827/apiserver.crt /tmp/978260827/expired/apiserver.crt: permission denied

https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/76534/pull-kubernetes-bazel-test/1117622723275657217

Reply:
looks like, this unit test is not written properly.
unit tests need to be called as non-root and this test tries to move files that one would expect to require root permissions.

what we really need to do is toggle off the public bits.
i'm indifferent on the change from 766 -> 750.

Reply:
@rosti feel free to LGTM this change if it seems ok for you.

Reply:
The 0660 omit the path traversal bit for a directory (the `x` bit), hence for non-root users this will fail even if you are the owner. Hence the test failure with 0660.
From my point of view, this is ok with 0750. However, I think that 0700 should work too and we should use that one, because of the possibility of storing a key there.

Reply:
yes, I tested 0700 work too, so let's use 0700.